### DISCUSSION TITLE
#GiveUpGitHub movement

### DISCUSSION THREAD

**Discussion by miurahr** [miurahr/py7zr#470] [General]
I'm afraid that py7zr project, as a GPL-ed free software project, depends on github,  and Software Freedom Conservancy advise to avoid Github for Free software project.

I urge you to read about the [Give up GitHub](https://GiveUpGitHub.org) campaign from [the Software Freedom Conservancy](https://sfconservancy.org) to understand some of the reasons why GitHub is not a good place to host FOSS projects.
  
I'd like to propose move py7zr project to other forge site which is built on free software.

There are several related project that have already  move the forge site, such as pyppmd, pybcj and inflate64.

**Comment by CBusse** [comment]
I would totally follow your project to any new place.

The points made by the SFC are sadly all valid complaints and  good reasons to move rather sooner than later.

But I have no real idea which alternative makes the most sense for the project, none of the proposals on the linked page feel good to me except for the self hosting solutions.

**Reply by miurahr** [reply]
I've selected codeberg.org for my main forge site.

You can see several python library projects on it.
- https://codeberg.org/miurahr/pyppmd
- https://codeberg.org/miurahr/inflate64

Because a CI/CD feature on codeberg.org is under development, I still use azure-pipelines as an alternative of github actions.
I can deliver python package for linux, macOS, and windows for amd64 architecture, but lacks aarch64(arm64) support on it.
Azure pipelines does not support YAML definition for the site other than github, gitbucket and gitlab, so I have been forced to use annoying web UI.

py7zr CI script checks on linux, macOS and windows, and arm64, with 3.6,3.7,3.8.3.9.3.10,3.11, pypy-3.8.
It will be able to port to azure-pipelines.
